src/transform/forEach.js

```diff
@@ -38,43 +37,72 @@ export default function(ast, logger) {
   });
 }
 
-function breakUsedInBody({body}) {
-  return traverser.find(body, node => {
-    return node.type === 'BreakStatement';
-  });
-}
+// Find if a statement is used in the for loop body,
+// skipping nested for loops and switch statements
+// Returns the line of the statement (if found);
+function statementUsedInBody({body}, statementType) {
+  let used;
+
+  traverser.traverse(body, {
+    enter(node) {
+      if (
+        node.type === 'SwitchStatement' ||
+        node.type === 'ForStatement'
```

Comment:
Ah yes, that does complicate things. I'll try both simplify this code and also make it handle every one of these cases.